File: Common/Production/FactoryDefinition.cs
using System.Collections.Generic;
using Epam.ImitationGames.Production.Common.Base;

namespace Epam.ImitationGames.Production.Common.Production
{
    /// <summary>
    /// Общее описание фабрики.
    /// </summary>
    public class FactoryDefinition : BaseProduction, IVisibleEntity
    {
        /// <summary>
        /// Тип производства.
        /// </summary>
        public ProductionType ProductionType { get; set; }

        /// <summary>
        /// Что умеет производить фабрика.
        /// </summary>
        /// <remarks>Ключ - уровень производства фабрики, Value - что фабрика умеет производить на этому уровне.</remarks>
        public Dictionary<int, List<Material>> CanProductionMaterials { get; set; }

        /// <summary>
        /// Базовое количество рабочих на фабрике, что бы она имела 100% производительность.
        /// </summary>
        public int BaseWorkers { get; set; }

        /// <summary>
        /// Уровень поколения фабрики.
        /// </summary>
        public int GenerationLevel { get; set; }

        /// <summary>
        /// Название производства.
        /// </summary>
        public string Name { get; set; }

        public string DisplayName => $"Описание производства {Name}, область {ProductionType.DisplayName}, поколение {GenerationLevel}, базовое количество рабочих {BaseWorkers}";
    }
}

File: src/IM.Production.UI/src/models/Production/FactoryDefinition.ts
import { ProductionType } from '../Production/ProductionType';
import { BaseProduction } from './BaseProduction';
import { IVisibleEntity } from '../Base/IVisibleEntity';

export class FactoryDefinition extends BaseProduction implements IVisibleEntity {
    constructor(
        /// <summary>
        /// Тип производства.
        /// </summary>
        public productionType: ProductionType,
        /// <summary>
        /// Что умеет производить фабрика.
        /// </summary>
        /// <remarks>Ключ - уровень производства фабрики, Value - что фабрика умеет производить на этому уровне.</remarks>
        public canProductionMaterials: any,
        /// <summary>
        /// Базовое количество рабочих на фабрике, что бы она имела 100% производительность.
        /// </summary>
        public baseWorkers: number,
        /// <summary>
        /// Уровень поколения фабрики.
        /// </summary>
        public generationLevel: number,
        /// <summary>
        /// Название производства.
        /// </summary>
        public name: string,
        public displayName: string,
        public id: any
    )
    {
        super(id);
    }
}